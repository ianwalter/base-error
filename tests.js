const { test } = require('@ianwalter/bff')
const BaseError = require('.')

test('Error message gets captured', ({ expect }) => {
  const msg = 'Bad Request'
  const error = new BaseError(new Error(msg))
  expect(error.message).toBe(msg)
})

test('String message gets captured as Error message', ({ expect }) => {
  const msg = 'Bad Request'
  const error = new BaseError(msg)
  expect(error.message).toBe(msg)
})

test('Error stacktrace gets captured', ({ expect }) => {
  try {
    throw new Error('Unparseable')
  } catch (err) {
    const baseError = new BaseError(err)
    const stackLines = baseError.stack.split('\n')
    expect(stackLines[0]).toBe('Error: Unparseable')
    expect(stackLines[1]).toContain('tests.js:18:11')
  }
})

test('String stacktrace gets captured', ({ expect }) => {
  const error = new BaseError('Undefined')
  const stackLines = error.stack.split('\n')
  expect(stackLines[0]).toBe('BaseError: Undefined')
  expect(stackLines[1]).toContain('tests.js:28:17')
})

test('Class extension', ({ expect }) => {
  class CustomError extends BaseError {
    constructor (err, code) {
      super(err)
      this.code = code
    }
  }

  const message = 'SyntaxError'
  const code = 1
  const error = new CustomError(message, code)
  const stackLines = error.stack.split('\n')
  expect(stackLines[0]).toBe(`CustomError: ${message}`)
  expect(stackLines[1]).toContain('tests.js:44:17')
  expect(error.message).toBe(message)
  expect(error.code).toBe(code)
})

test('Multiple parameter handling', ({ expect }) => {
  const msg = 'Bad Request'
  const arg2 = { thisis: { nested: true } }
  const arg3 = [1, 2, 3]
  const err = new BaseError(msg, arg2, arg3)
  expect(err.message)
    .toBe(`${msg}\n${JSON.stringify(arg2)}\n${JSON.stringify(arg3)}`)
})
