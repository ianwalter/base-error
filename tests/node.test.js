/**
 * @jest-environment node
 */
require('./tests')
