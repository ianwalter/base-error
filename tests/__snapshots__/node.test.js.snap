// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseError should be able to be extended to create a custom Error 1`] = `
"CustomError: SyntaxError
    at Object.<anonymous>.test (/Users/iwalter/ianwalter/base-error/tests/tests.js:39:17)
    at Object.asyncJestTest (/Users/iwalter/ianwalter/base-error/node_modules/jest-jasmine2/build/jasmine_async.js:108:37)
    at resolve (/Users/iwalter/ianwalter/base-error/node_modules/jest-jasmine2/build/queue_runner.js:56:12)
    at new Promise (<anonymous>)
    at mapper (/Users/iwalter/ianwalter/base-error/node_modules/jest-jasmine2/build/queue_runner.js:43:19)
    at promise.then (/Users/iwalter/ianwalter/base-error/node_modules/jest-jasmine2/build/queue_runner.js:87:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)"
`;

exports[`BaseError should capture the stacktrace if given an Error 1`] = `
"Error: Unparseable
    at Object.<anonymous>.test (/Users/iwalter/ianwalter/base-error/tests/tests.js:17:11)
    at Object.asyncJestTest (/Users/iwalter/ianwalter/base-error/node_modules/jest-jasmine2/build/jasmine_async.js:108:37)
    at resolve (/Users/iwalter/ianwalter/base-error/node_modules/jest-jasmine2/build/queue_runner.js:56:12)
    at new Promise (<anonymous>)
    at mapper (/Users/iwalter/ianwalter/base-error/node_modules/jest-jasmine2/build/queue_runner.js:43:19)
    at promise.then (/Users/iwalter/ianwalter/base-error/node_modules/jest-jasmine2/build/queue_runner.js:87:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)"
`;

exports[`BaseError should create a stacktrace if given a string 1`] = `
"BaseError: Undefined
    at Object.<anonymous>.test (/Users/iwalter/ianwalter/base-error/tests/tests.js:25:17)
    at Object.asyncJestTest (/Users/iwalter/ianwalter/base-error/node_modules/jest-jasmine2/build/jasmine_async.js:108:37)
    at resolve (/Users/iwalter/ianwalter/base-error/node_modules/jest-jasmine2/build/queue_runner.js:56:12)
    at new Promise (<anonymous>)
    at mapper (/Users/iwalter/ianwalter/base-error/node_modules/jest-jasmine2/build/queue_runner.js:43:19)
    at promise.then (/Users/iwalter/ianwalter/base-error/node_modules/jest-jasmine2/build/queue_runner.js:87:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)"
`;
